department-of-veterans-affairs/abd-vro
# Vro Process Completion Slack Messages

## What was the problem?


There are a number of edge cases during the last step of  VRO where the claim is updated that we wanted to send slack messages.

1. If the Temp Station of Jurisdiction changes during processing. Although Temp Station of Jurisdiction is checked at the start of the processing to be 398, historically we also verified it at the end of the processing to double check. Some of those checks are removed recently. We still had a audit entry if this happens. This removes others as well and just sends a slack message if Temp Station of Jurisdiction has changed something other than 398 during processing.
2. We check BIP Claims API being down at the start of processing. At the end of processing we assume it is functional and do not really handle any errors that might happen during updating of the claim.

Associated tickets or Slack threads:

- [MCP-2604](https://amida.atlassian.net/browse/MCP-2604)
- [MCP-2605](https://amida.atlassian.net/browse/MCP-2605)

## How does this fix it?[^1]


1. We now  send a slack message as well as storing n audit trail if Temp Station of Jurisdiction has changed something other than 398 during processing.
2. We now handle BIP Claims API related error during update  of the claim at the end of processing and send a slack message when problems.

## How to test this PR
- Two end-2-end tests are added for testing. Make sure they pass and check if they are testing what is specified above.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
